core/interval/int_test.go

```diff
@@ -0,0 +1,288 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package interval_test
+
+import (
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/core/interval"
+)
+
+type integerIntervalSuite struct{}
+
+var _ = gc.Suite(&integerIntervalSuite{})
+
+func (s *integerIntervalSuite) TestNewIntegerInterval(c *gc.C) {
+	i := interval.NewIntegerInterval(1, 5)
+	c.Check(i.Lower, gc.Equals, 1)
+	c.Check(i.Upper, gc.Equals, 5)
+
+	i = interval.NewIntegerInterval(5, 1)
+	c.Check(i.Lower, gc.Equals, 1)
+	c.Check(i.Upper, gc.Equals, 5)
+}
+
+func (s *integerIntervalSuite) TestIntersects(c *gc.C) {
```

Comment:
Would be good to have some serious edge case testing for all of these. i.e. does (0,0) intersect with (0,0), And things like that.

Reply:
I feel these tests do mostly have good coverage in most places. I have added a number of additional test where they are lacking